core/src/CMakeLists.txt

```diff
@@ -113,6 +112,45 @@ IF (Kokkos_ENABLE_IMPL_DESUL_ATOMICS AND NOT desul_FOUND)
   )
 ENDIF()
 
+IF (KOKKOS_ENABLE_IMPL_MDSPAN)
+  MESSAGE(STATUS "Experimental mdspan support is enabled")
+
+  # Some compilers now include mdspan... we just flag on their version
+  # for now until we can get some compiler detection support
+  include(CheckIncludeFileCXX)
+  check_include_file_cxx(experimental/mdspan KOKKOS_COMPILER_SUPPORTS_EXPERIMENTAL_MDSPAN)
+  check_include_file_cxx(mdspan KOKKOS_COMPILER_SUPPORTS_MDSPAN)
+
+  if (KOKKOS_ENABLE_MDSPAN_EXTERNAL)
+    MESSAGE(STATUS "Using external mdspan")
+    target_link_libraries(kokkoscore PUBLIC std::mdspan)
+  elseif(KOKKOS_COMPILER_SUPPORTS_MDSPAN)
+    message(STATUS "Using compiler-supplied mdspan")
+  elseif(KOKKOS_COMPILER_SUPPORTS_EXPERIMENTAL_MDSPAN)
+    message(STATUS "Using compiler-supplied experimental/mdspan")
+  else()
+    KOKKOS_LIB_INCLUDE_DIRECTORIES(
+      kokkoscore
+      ${CMAKE_CURRENT_SOURCE_DIR}/../../tpls/mdspan/include
+    )
```

Comment:
So this means we have no way of forcing using the mdspan bundled if the compiler sees the header.
Is that intentional?

Reply:
We probably gonna need one eventually assuming that it does not provide enough stuff.